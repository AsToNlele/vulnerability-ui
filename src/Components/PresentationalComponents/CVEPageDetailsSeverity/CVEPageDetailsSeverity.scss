@import "../../../App";

.cve-cvss-base-score {
    border-bottom: 1px solid var(--pf-global--BorderColor--100);
    display: flex;
    align-items: baseline;

    > h1 {
        margin-right: var(--pf-global--spacer--sm) !important;
    }
    > h4 {
        margin-top: 0px !important;
    }
}

.cve-cvss-base-score-value {
    font-weight: bold;
    padding-right: 0.5em;
}

.cve-impact-badge {
    width: 1.8em;
}
