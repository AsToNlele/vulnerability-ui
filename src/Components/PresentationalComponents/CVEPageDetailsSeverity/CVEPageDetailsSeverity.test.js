import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CVEPageDetailsSeverity from './CVEPageDetailsSeverity';

describe('CVEPageDetailsSeverity component', () => {
    it('should render without props', () => {
        const wrapper = shallow(<CVEPageDetailsSeverity />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render where impact value is undefined', () => {
        const wrapper = shallow(<CVEPageDetailsSeverity impact={undefined} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render where impact value is Critial', () => {
        const wrapper = shallow(<CVEPageDetailsSeverity impact={'Critical'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
