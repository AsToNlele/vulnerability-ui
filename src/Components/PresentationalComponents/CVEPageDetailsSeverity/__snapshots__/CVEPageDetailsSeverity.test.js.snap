// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEPageDetailsSeverity component should render where impact value is Critial 1`] = `
<Fragment>
  <Stack
    className=""
    component="div"
    gutter="sm"
  >
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className="cve-cvss-base-score"
      >
        <Text
          className=""
          component="h1"
        >
          N/A
        </Text>
        <Text
          className=""
          component="h4"
        >
          CVSS base score
        </Text>
      </TextContent>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <Text
          className=""
          component="h6"
        >
          Severity
        </Text>
      </TextContent>
      <Split
        className=""
        component="div"
        gutter={null}
      >
        <SplitItem
          className=""
          isMain={false}
        >
          <Shield
            hasTooltip={false}
            impact="Critical"
            size="md"
            title=""
            tooltipPosition="top"
            tooltipPrefix=""
          />
        </SplitItem>
        <SplitItem
          className=""
          isMain={false}
        >
          <Title
            size="md"
          >
            <SeverityRating
              format="title"
              impact="Critical"
            />
          </Title>
        </SplitItem>
      </Split>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <SeverityRating
          format="link"
          impact="Critical"
        />
      </TextContent>
    </StackItem>
  </Stack>
</Fragment>
`;

exports[`CVEPageDetailsSeverity component should render where impact value is undefined 1`] = `
<Fragment>
  <Stack
    className=""
    component="div"
    gutter="sm"
  >
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className="cve-cvss-base-score"
      >
        <Text
          className=""
          component="h1"
        >
          N/A
        </Text>
        <Text
          className=""
          component="h4"
        >
          CVSS base score
        </Text>
      </TextContent>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <Text
          className=""
          component="h6"
        >
          Severity
        </Text>
      </TextContent>
      <Split
        className=""
        component="div"
        gutter={null}
      >
        <SplitItem
          className=""
          isMain={false}
        >
          <Shield
            hasTooltip={false}
            impact="Unknown"
            size="md"
            title=""
            tooltipPosition="top"
            tooltipPrefix=""
          />
        </SplitItem>
        <SplitItem
          className=""
          isMain={false}
        >
          <Title
            size="md"
          >
            <SeverityRating
              format="title"
              impact="Unknown"
            />
          </Title>
        </SplitItem>
      </Split>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <SeverityRating
          format="link"
          impact="Unknown"
        />
      </TextContent>
    </StackItem>
  </Stack>
</Fragment>
`;

exports[`CVEPageDetailsSeverity component should render without props 1`] = `
<Fragment>
  <Stack
    className=""
    component="div"
    gutter="sm"
  >
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className="cve-cvss-base-score"
      >
        <Text
          className=""
          component="h1"
        >
          N/A
        </Text>
        <Text
          className=""
          component="h4"
        >
          CVSS base score
        </Text>
      </TextContent>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <Text
          className=""
          component="h6"
        >
          Severity
        </Text>
      </TextContent>
      <Split
        className=""
        component="div"
        gutter={null}
      >
        <SplitItem
          className=""
          isMain={false}
        >
          <Shield
            hasTooltip={false}
            impact="Unknown"
            size="md"
            title=""
            tooltipPosition="top"
            tooltipPrefix=""
          />
        </SplitItem>
        <SplitItem
          className=""
          isMain={false}
        >
          <Title
            size="md"
          >
            <SeverityRating
              format="title"
              impact="Unknown"
            />
          </Title>
        </SplitItem>
      </Split>
    </StackItem>
    <StackItem
      className=""
      isMain={false}
    >
      <TextContent
        className=""
      >
        <SeverityRating
          format="link"
          impact="Unknown"
        />
      </TextContent>
    </StackItem>
  </Stack>
</Fragment>
`;
